### PR TITLE
1920: Update admin field names

### LINKED ISSUE #1920: Django admin: Update field labels
### Issue description

We audited field labels across the admin and identified opportunities to make the labels more meaningful and consistent. 

Also, we audited the search bar "helper text" (which appears below each search bar in the admin). We need to make a few tweaks to those, as well.

### Acceptance criteria

- [x] Update field labels within the admin as outlined within the [updates to labels](https://docs.google.com/spreadsheets/d/1dXe28ZpyghFpxB_T-GZsDpFUKb96AdztsNPhFL0BvpE/edit#gid=393682108) tab of the spreadsheet. 
- [x] Update the search bar "helper text" as outlined with the [search helper](https://docs.google.com/spreadsheets/d/1dXe28ZpyghFpxB_T-GZsDpFUKb96AdztsNPhFL0BvpE/edit#gid=1935823459) tab of the spreadsheet.
- [x] Applies to Analysts and Super Users.

### Additional context

Reach out to Katherine if there are any questions about the spreadsheet.
### Links to other issues

Related to #1946

### PR DESCRIPTION
**Pull request by erinysong**
## Ticket

Resolves #1920 

## Changes

- Updates admin table and form field names as requested in [this spreadsheet](https://docs.google.com/spreadsheets/d/1dXe28ZpyghFpxB_T-GZsDpFUKb96AdztsNPhFL0BvpE/edit#gid=393682108)
- Updates search help text as requested in [this spreadsheet](https://docs.google.com/spreadsheets/d/1dXe28ZpyghFpxB_T-GZsDpFUKb96AdztsNPhFL0BvpE/edit#gid=1935823459).

## Setup

Go to [getgov-es sandbox admin](https://getgov-es.app.cloud.gov/admin/) and verify all content changes are implemented.

#### Satisfied acceptance criteria and met development standards

- [ ] Met the acceptance criteria, or will meet them in a subsequent PR
- [ ] Created/modified automated tests
- [ ] Added at least 2 developers as PR reviewers (only 1 will need to approve)
- [ ] Messaged on Slack or in standup to notify the team that a PR is ready for review
- [ ] Changes to “how we do things” are documented in READMEs and or onboarding guide
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Original Developer)

- [ ] All new functions and methods are commented using plain language
- [ ] Did dependency updates in Pipfile also get changed in requirements.txt?
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values

#### Validated user-facing changes (if applicable)

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)
- [ ] Add at least 1 designer as PR reviewer

#### Reviewed, tested, and left feedback about the changes

- [ ] Pulled this branch locally and tested it
- [ ] Reviewed this code and left comments
- [ ] Checked that all code is adequately covered by tests
- [ ] Made it clear which comments need to be addressed before this work is merged
- [ ] If any model was updated to modify/add/delete columns, makemigrations was ran and the associated migrations file has been commited.

#### Ensured code standards are met (Code reviewer)

- [ ] All new functions and methods are commented using plain language
- [ ] Interactions with external systems are wrapped in try/except
- [ ] Error handling exists for unusual or missing values
- [ ] (Rarely needed) Did dependency updates in Pipfile also get changed in requirements.txt?

#### Validated user-facing changes as a developer

- [ ] New pages have been added to .pa11yci file so that they will be tested with our automated accessibility testing
- [ ] Checked keyboard navigability
- [ ] Meets all designs and user flows provided by design/product
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers, the suggestion is to use ones that the developer didn't (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

**Note:** Multiple code reviewers can share the checklists above, a second reviewers should not make a duplicate checklist

#### Verified that the changes match the design intention

- [ ] Checked that the design translated visually
- [ ] Checked behavior
- [ ] Checked different states (empty, one, some, error)
- [ ] Checked for landmarks, page heading structure, and links
- [ ] Tried to break the intended flow

#### Validated user-facing changes as a designer

- [ ] Checked keyboard navigability
- [ ] Tested general usability, landmarks, page header structure, and links with a screen reader (such as Voiceover or ANDI)

- [ ] Tested with multiple browsers (check off which ones were used)
  - [ ] Chrome
  - [ ] Microsoft Edge
  - [ ] FireFox
  - [ ] Safari

- [ ] (Rarely needed) Tested as both an analyst and applicant user

### CODE REVIEW

File: src/registrar/models/domain_information.py [hunk 1]
```diff
@@ -108,6 +109,7 @@ class DomainInformation(TimeStampedModel):
     is_election_board = models.BooleanField(
         null=True,
         blank=True,
+        verbose_name="Election office",
         help_text="Is your organization an election office?",
```
Comment (erinysong) [inline src/registrar/models/domain_information.py:113]: Does anyone know why we have 2 mentions of setting the field is_election_board?

**Comment by Katherine-Osos** [comment]
@erinysong This looks so great! Just one edit: 
- In the **Draft Domains** table, change the column heading "Draft domain" to "Requested domain"

File: src/registrar/models/domain_information.py [hunk 2]
```diff
@@ -177,6 +180,7 @@ class DomainInformation(TimeStampedModel):
         null=True,
         # Access this information via Domain as "domain.domain_info"
         related_name="domain_info",
+        verbose_name="domain request",
```
Comment (therealslimhsiehdy) [inline src/registrar/models/domain_information.py:183]: **Question:** Why is the verbose name domain request versus domain info 😅

Comment (erinysong) [reply src/registrar/models/domain_information.py]: 😅 Also an error on my part - just removed this and thank you for catching this!

**Review by therealslimhsiehdy — APPROVED** [review]
LGTM -- Erin and I went through each update in a meeting to double check each label and search helper text to confirm the visual changes.

**Review by Katherine-Osos — APPROVED** [review]
LGTM!

**Pull request merged**